scripts/validate-build.js

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env node
+// eslint-disable-next-line strict
+'use strict';
+
+const fs = require('fs');
+const esprima = require('esprima');
+const walk = require('esprima-walk');
+const colors = require('colors');
+
+/**
+ * Load the Abstract Syntax Tree from the javascript file given
+ *
+ * @param {String} file - The path to the javascript filename
+ * @returns {ASTTree} Returns the AST tree
+ */
+function loadAST(file) {
+  try {
+    return esprima.parse(
+      String(fs.readFileSync(file)),
+      { comment: true }
+    );
+
+  } catch (e) {
+    console.error('Unable to parse source:', e);
+
+    return false;
+  }
+}
+
+/**
+ * Collect all comments from the AST
+ *
+ * @param {ASTTree} ast - The AST tree
+ * @returns {Array<ASTNode>} Returns the comment AST nodes
+ */
+function loadComments(ast) {
+  let comments = [];
+
+  walk(ast, function (node) {
```

Comment:
You do not need to traverse the AST to collect the comments. Every single comment is available from the top-level object already.
For more details, read http://esprima.readthedocs.io/en/3.1/syntactic-analysis.html#comment-collection.